prov/sockets/include/sock.h

```diff
@@ -1211,6 +1218,14 @@ int sock_epoll_wait(struct sock_epoll_set *set, int timeout);
 int sock_epoll_get_fd_at_index(struct sock_epoll_set *set, int index);
 void sock_epoll_close(struct sock_epoll_set *set);
 
+int sock_epoll_map_init(struct sock_epoll_map *emap);
+int sock_epoll_map_destroy(struct sock_epoll_map *emap);
+int sock_epoll_map_add(struct sock_epoll_map *emap, int fd, void *ptr);
+int sock_epoll_map_del(struct sock_epoll_map *emap, int fd);
+void *sock_epoll_map_lookup(struct sock_epoll_map *emap, int fd);
+int sock_epoll_map_signal(struct sock_epoll_map *emap);
+int sock_epoll_map_wait(struct sock_epoll_map *emap);
```

Comment:
See include/fi_signal.h and src/common.c.  The fi_epoll abstraction should be providing the functionality that you need.  sock_epoll.c should be removed and replaced with the general implementation.